### PR TITLE
[DOCS] Legacy MO Extensibility update for master

### PR DESCRIPTION
**Pull request by msmykx-intel**
* Divided MO Extensibility article into separate smaller articles,
* Applied the suggestion from https://github.com/openvinotoolkit/openvino/pull/15679
* Recreated images in svg format
* Fixing directives

### CODE REVIEW

**Comment by rkazants** [comment]
@msmykx-intel, please let us review before merge:)
cc, @slyalin

File: docs/MO_DG/prepare_model/Model_Optimizer_FAQ.md [hunk 1]
```diff
@@ -2,6 +2,12 @@
 
 If your question is not covered by the topics below, use the [OpenVINO Support page](https://software.intel.com/en-us/openvino-toolkit/documentation/get-started), where you can participate on a free forum.
 
+@sphinxdirective
+
+.. _mo_question_1:
+
+@endsphinxdirective
```
Comment (rkazants) [inline docs/MO_DG/prepare_model/Model_Optimizer_FAQ.md:9]: can you please clarify for what these changes are? Is it only for the first question?

File: docs/MO_DG/prepare_model/customize_model_optimizer/Customize_Model_Optimizer.md [hunk 2]
```diff
@@ -6,217 +6,218 @@
...
    openvino_docs_MO_DG_prepare_model_customize_model_optimizer_Extending_Model_Optimizer_With_Caffe_Python_Layers
 
-@endsphinxdirective
-
-
 This article describes Model Optimizer internals. Altering them may result in application instability, and in case of future changes to the API, lack of backward compatibility.
 
-> **NOTE**: If you want to add support for ONNX, PaddlePaddle or Tensorflow operations, or you are not familiar with other extension alternatives in OpenVINO, read [this guide](../../../Extensibility_UG/Intro.md) instead.
+.. note:: 
+   If you want to add support for ONNX, PaddlePaddle or Tensorflow operations, or you are not familiar with other extension alternatives in OpenVINO, read :doc:`this guide <openvino_docs_Extensibility_UG_Intro>` instead.
```
Comment (rkazants) [inline docs/MO_DG/prepare_model/customize_model_optimizer/Customize_Model_Optimizer.md:16]:
```suggestion
   If you want to add support for ONNX, PaddlePaddle or TensorFlow operations, or you are not familiar with other extension alternatives in OpenVINO, read :doc:`this guide <openvino_docs_Extensibility_UG_Intro>` instead.
```

File: docs/MO_DG/prepare_model/customize_model_optimizer/Customize_Model_Optimizer.md [hunk 3]
```diff
@@ -6,217 +6,218 @@
...
+
+.. _model-optimizer-extensibility:
 
-<a name="model-optimizer-extensibility"></a>Model Optimizer extensibility mechanism enables support of new operations and custom transformations to generate the optimized intermediate representation (IR) as described in the
-[Deep Learning Network Intermediate Representation and Operation Sets in OpenVINO™](../../IR_and_opsets.md). This
-mechanism is a core part of Model Optimizer, as a huge set of examples showing how to add custom logic to support your model.
+Model Optimizer extensibility mechanism enables support of new operations and custom transformations to generate the optimized intermediate representation (IR) as described :doc:`here <openvino_docs_MO_DG_IR_and_opsets>`. This mechanism is a core part of Model Optimizer, as a huge set of examples showing how to add custom logic to support your model.
```
Comment (rkazants) [inline docs/MO_DG/prepare_model/customize_model_optimizer/Customize_Model_Optimizer.md:20]:
```suggestion
Model Optimizer extensibility mechanism enables support of new operations and custom transformations to generate the optimized intermediate representation (IR) as described :doc:`here <openvino_docs_MO_DG_IR_and_opsets>`.
This mechanism is a core part of Model Optimizer, as a huge set of examples showing how to add custom logic to support your model.
```

Comment (rkazants) [reply docs/MO_DG/prepare_model/customize_model_optimizer/Customize_Model_Optimizer.md:20]: I think better to split long lines. The same comment below

File: docs/MO_DG/prepare_model/customize_model_optimizer/Customize_Model_Optimizer.md [hunk 4]
```diff
@@ -6,217 +6,218 @@
    :maxdepth: 1
    :hidden:
```
Comment (rkazants) [inline docs/MO_DG/prepare_model/customize_model_optimizer/Customize_Model_Optimizer.md:8]: Now you are adding `Model Optimizer Extensibility` for legacy into `OpenVINO Extensibility` part. In `OpenVINO Extensibility` we should have only extensibility logic for new frontends.

File: docs/MO_DG/prepare_model/customize_model_optimizer/Customize_Model_Optimizer.md [hunk 5]
```diff
@@ -6,217 +6,218 @@
...
 
+   openvino_docs_MO_DG_prepare_model_customize_model_optimizer_Customize_Model_Optimizer_Model_Optimizer_Ports_Connections
+   openvino_docs_MO_DG_prepare_model_customize_model_optimizer_Model_Optimizer_Extensions
    openvino_docs_MO_DG_prepare_model_customize_model_optimizer_Extending_Model_Optimizer_With_Caffe_Python_Layers
```
Comment (rkazants) [inline docs/MO_DG/prepare_model/customize_model_optimizer/Customize_Model_Optimizer.md:12]: I think we need some section in `[Model Optimizer Usage](http://openvino-doc.iotg.sclab.intel.com/smyk-mo-extensibility-rst-test-final-2/openvino_docs_MO_DG_Deep_Learning_Model_Optimizer_DevGuide.html)` about MO extensibility or to have some link.

I don' know where to locate the legacy approach to extend. @slyalin, please advice.

**Review by rkazants — CHANGES_REQUESTED** [review]
Not recommended to merge because we should not change documentation about the legacy approach. We just need to locate it properly into documentation structure.
Also, I believe we need some small section or link to MO extensibility in `[Model Optimizer Usage](http://openvino-doc.iotg.sclab.intel.com/smyk-mo-extensibility-rst-test-final-2/openvino_docs_MO_DG_Deep_Learning_Model_Optimizer_DevGuide.html#)`

File: docs/MO_DG/prepare_model/customize_model_optimizer/Customize_Model_Optimizer.md [hunk 6]
```diff
@@ -6,217 +6,218 @@
...
 
-@endsphinxdirective
-
-
 This article describes Model Optimizer internals. Altering them may result in application instability, and in case of future changes to the API, lack of backward compatibility.
```
Comment (slyalin) [inline docs/MO_DG/prepare_model/customize_model_optimizer/Customize_Model_Optimizer.md:13]: I don't understand why we are investing to modifying and better organizing internal part of the documentation for the legacy MO infrastructure? With migrating to the new Tensorflow frontend we don't need to propose extensibility framework based on old MO extensions for users. Everything will be moved to the new extensions described in a dedicated part. This one should be deprecated soon. Please prepare a move of this whole document to developer documentation with appropriate legacy statement rather then keeping and updating it here. The main document for extensibility that everybody should extend and contribute to is https://docs.openvino.ai/latest/openvino_docs_Extensibility_UG_Frontend_Extensions.html. It works for both MO and runtime, this is a single extension API for OpenVINO. The only thing that prevented us from more aggressive removing of this legacy part was Tensorflow. Now we are migrating to the new one and this legacy part is obsolete.

**Review by slyalin — CHANGES_REQUESTED** [review]
Provide more aggressive statement that this is a legacy API and soon will be abandoned in favor of new OpenVINO extensions described in a dedicated chapter.

**Comment by msmykx-intel** [comment]
Added Legacy tag for the old extension guide along with warning note at the beginning of the article, to inform user about new TF extension and leads to Frontend Extensions.

Legacy tags were also added in cross-links in other parts of documentation.

Also added link to OpenVINO Extensibility Mechanism introduction in Model Optimizer Usage.

File: docs/MO_DG/prepare_model/customize_model_optimizer/Customize_Model_Optimizer.md (hunk 5 above)
Comment (msmykx-intel) [reply docs/MO_DG/prepare_model/customize_model_optimizer/Customize_Model_Optimizer.md:12]: Done in https://github.com/openvinotoolkit/openvino/pull/15931/#issuecomment-1499038492

File: docs/MO_DG/prepare_model/customize_model_optimizer/Customize_Model_Optimizer.md [hunk 7]
```diff
@@ -1,222 +1,228 @@
-# Model Optimizer Extensibility {#openvino_docs_MO_DG_prepare_model_customize_model_optimizer_Customize_Model_Optimizer}
+# [LEGACY] Model Optimizer Extensibility {#openvino_docs_MO_DG_prepare_model_customize_model_optimizer_Customize_Model_Optimizer}
```
Comment (slyalin) [inline docs/MO_DG/prepare_model/customize_model_optimizer/Customize_Model_Optimizer.md:1]: Still don't see a good explanation why this chapter is considered to be legacy. Please add something like described here: https://docs.openvino.ai/latest/openvino_docs_Extensibility_UG_Intro.html#mapping-from-framework-operation about why two extensibility exists. Focus on explaining that the code described here will be deprecated soon and kept only for backward compatibility and old users. I'm not sure about precise words, but it should clearly stops new users to continue reading this document.

**Review by slyalin — APPROVED** [review]

**Review by rkazants — APPROVED** [review]

**Review by kblaszczak-intel — APPROVED** [review]

**Pull request merged**